src/providers/EtherspotContextProvider.tsx

```diff
@@ -540,6 +547,51 @@ const EtherspotContextProvider = ({
     [sdk, accountAddress]
   );
 
+  const getTransactionsFromChain = useCallback(
+    async (chainId: number, address: string | null): Promise<Transactions[]> => {
+      const sdk = getSdkForChainId(chainId);
+      const account = address;
+      if (!!sdk && !account) return [];
+
+      await sdk.computeContractAccount();
+      try {
+        let transactions: any = [];
+        const txs = await sdk.getTransactions({ account });
+        if (txs?.items) transactions = txs.items?.map((item) => ({ ...item, chainId }));
+        return { transactions };
+      } catch (e) {
+        //
```

Comment:
Please assume that `sdk.getTransactions` will fail - what happens in this catch blocl?

Reply:
if (txs?.items) transactions = txs.items?.map((item) => ({ ...item, chainId }));
        return { transactions }; 
        return []; in same function, as I am using callback function

Reply:
Ex:  const getEnsNode = useCallback(
    async (
      chainId: number,
      address: string | null = accountAddress,
      recompute: boolean = true
    ): Promise<ENSNode | null> => {
      const sdkForChain = getSdkForChainId(chainId);

      if (!sdkForChain) return null;

      let computedAccount;

      if (!address && recompute) {
        try {
          computedAccount = await connect();
        } catch (e) {
          //
        }
      }

      if (!address && !computedAccount) return null;

      try {
        const ens = await sdkForChain.getENSNode({
          nameOrHashOrAddress: address || computedAccount,
        });
        return ens;
      } catch (e) {
        //
      }

      return null;
    },
    [sdk, accountAddress]
  );

Reply:
Understood that the transaction has a fallback, but you're assuming that the SDK will not throw an error if the following may occur:

- Network issue
- Internal server errors
- SDK internal error

In any of these scenarios, the catch block will fire - where there's nothing happening. You should at least let the user know that something is wrong and they can try again.